Movies_MVVM/Core/NetworkService/Protocols/NetworkServiceCoreProtocol.swift

```diff
@@ -0,0 +1,14 @@
+// NetworkServiceCoreProtocol.swift
+// Copyright © Aleksandr Dorofeev. All rights reserved.
+
+import Foundation
+
+/// Network service core protocol
+protocol NetworkServiceCoreProtocol {
+    func getJson<T: Decodable>(
+        id: String?,
+        categoryOfMovies: String?,
+        page: Int?,
```

Comment:
```suggestion
        id: String,
        categoryOfMovies: String,
        page: Int,
```
Опционалы наверное потому, что они возвращаются из модели, но метод который принимает опционалы предполагает работу как с нормальными значениями, так и с nil. Наверное нужно их разворачивать до вызова метода, а не внутри. Это не замечание, просто рассуждение

Reply:
Это опционалы, потому что метод универсальный, я где то передаю параметр где то нет